### PR TITLE
WEA-25 Move all temperature conversion logic to repository

### PR DESCRIPTION
**Pull request by ayush-shta**
1. Move all temperature conversion logic to the repository.
2. Remove change background function from home contract.
3. Refactored all temperatures in Entity models to type int instead of double.

@subash-poudel

### CODE REVIEW

File: app/src/main/java/com/ayush/weatherapp/entities/CurrentForecastEntity.java [hunk 1]
```diff
@@ -30,11 +30,11 @@ public void setIcon(String icon) {
     this.icon = icon;
   }
 
-  public double getTemperature() {
+  public int getTemperature() {
     return temperature;
   }
 
   public void setTemperature(double temperature) {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/entities/CurrentForecastEntity.java:37]: The conversion can be done in the mapper.

File: app/src/main/java/com/ayush/weatherapp/entities/DailyDataEntity.java [hunk 2]
```diff
@@ -78,20 +78,20 @@ public void setSunsetTime(int sunsetTime) {
     this.sunsetTime = sunsetTime;
   }
 
-  public double getTemperatureHigh() {
+  public int getTemperatureHigh() {
     return temperatureHigh;
   }
 
   public void setTemperatureHigh(double temperatureHigh) {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/entities/DailyDataEntity.java:86]: from mappper

File: app/src/main/java/com/ayush/weatherapp/entities/DailyDataEntity.java [hunk 3]
```diff
@@ -78,20 +78,20 @@ public void setSunsetTime(int sunsetTime) {
...
   }
 
-  public double getTemperatureLow() {
+  public int getTemperatureLow() {
     return temperatureLow;
   }
 
   public void setTemperatureLow(double temperatureLow) {
-    this.temperatureLow = temperatureLow;
+    this.temperatureLow = (int) temperatureLow;
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/entities/DailyDataEntity.java:95]: .

File: app/src/main/java/com/ayush/weatherapp/entities/HourlyDataEntity.java [hunk 4]
```diff
@@ -54,12 +54,12 @@ public void setIcon(String icon) {
...
   }
 
-  public double getTemperature() {
+  public int getTemperature() {
     return temperature;
   }
 
   public void setTemperature(double temperature) {
-    this.temperature = temperature;
+    this.temperature = (int) temperature;
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/entities/HourlyDataEntity.java:62]: .

File: app/src/main/java/com/ayush/weatherapp/home/DailyForecastFragment.java [hunk 5]
```diff
@@ -51,17 +53,17 @@ public void setData(List<DailyDataEntity> dailyForecastList) {
...
 
   private void setView(DailyDataEntity dailyData) {
-    double averageTemperature = Math.round(
-        MathUtils.getAverage(dailyData.getTemperatureHigh(), dailyData.getTemperatureLow()));
-
     ForecastCompoundView forecastCompoundView =
         (ForecastCompoundView) getLayoutInflater().inflate(R.layout.item_forecast_compound_view,
             llForecastDetails, false);
 
+    long averageTemperature = Math.round(
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/DailyForecastFragment.java]: logic in presenter

File: app/src/main/java/com/ayush/weatherapp/home/HomeContract.java [hunk 6]
```diff
@@ -1,24 +1,22 @@
...
-import com.ayush.weatherapp.retrofit.weatherApi.pojo.DailyDataDTO;
-import com.ayush.weatherapp.retrofit.weatherApi.pojo.HourlyDataDTO;
 import java.util.List;
 
 public interface HomeContract {
   interface View extends BaseContract.BaseView {
     void setCurrentForecast(CurrentForecastEntity currentForecastDTO);
 
-    void setCurrentTemperature(String temperature);
+    void setCurrentTemperature(int temperature, @Temperature int tempUnit);
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeContract.java:14]: We could make this dumb by changing the parameter to String and calculating in presenter

File: app/src/main/java/com/ayush/weatherapp/home/HourlyForecastFragment.java [hunk 7]
```diff
@@ -53,12 +55,12 @@ private void setView(HourlyDataEntity hourlyData) {
         (ForecastCompoundView) getLayoutInflater().inflate(R.layout.item_forecast_compound_view,
             llForecastDetails, false);
 
-    double hourlyTemperature = hourlyData.getTemperature();
+    long hourlyTemperature = hourlyData.getTemperature();
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HourlyForecastFragment.java]: redundant

File: app/src/main/java/com/ayush/weatherapp/home/DailyForecastFragment.java (hunk 5 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/DailyForecastFragment.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HourlyForecastFragment.java (hunk 7 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HourlyForecastFragment.java]: :ok:

**Pull request merged**